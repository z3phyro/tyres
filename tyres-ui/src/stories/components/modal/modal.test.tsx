import { expect, test } from "vitest";
import { render, screen } from "@solidjs/testing-library";
import Modal from "./modal";

test("Modal renders", () => {
  window.scrollTo = () => { };
  render(() => <Modal title="" description=""></Modal>);

  const modal = screen.getByRole("dialog");
  expect(modal).toBeTruthy();
});

test("Modal renders with custom content", () => {
  window.scrollTo = () => { };
  render(() => <Modal title="Modal title" description="Modal description" />);

  const title = screen.getByRole("heading");
  expect(title.textContent).toBe("Modal title");

  const description = screen.getByRole("paragraph");
  expect(description.textContent).toBe("Modal description");
});

test("Modal snapshot test", () => {
  window.scrollTo = () => { };
  const { asFragment } = render(() => <Modal title="" description=""></Modal>);

  expect(asFragment()).toMatchSnapshot();
});
