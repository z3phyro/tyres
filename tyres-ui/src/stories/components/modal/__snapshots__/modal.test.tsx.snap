// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal snapshot test 1`] = `""`;
